robottelo/api/utils.py

```diff
@@ -437,3 +438,34 @@ def configure_provisioning(org=None, loc=None):
         'host_group': host_group.name,
         'domain': domain.name,
     }
+
+
+def get_role_by_bz(bz_id):
+    """Create and configure custom role entity for the testing of specific bugs
+     This function will read the dictionary of permissions and their associated
+     bugzilla id's from robottelo.constants "PERMISSIONS_WITH_BZ",
+     these permissions will create filter and a single role will be created
+     from all the filters.
+
+     :param bz_id: This is the bugzilla id that is specified in the
+        PERMISSIONS_WITH_BZ list, all the permissions associated with the bz_id
+        will be fetched and filters will be created
+     :return: A single role entity will be created from all the created filters
+     """
+    perms_dict = {}
+    role = entities.Role().create()
+    for res_type in PERMISSIONS_WITH_BZ.keys():
+        for value in PERMISSIONS_WITH_BZ[res_type]:
+            if bz_id in value.get('bz', []):
+                if res_type in perms_dict:
+                    perms_dict[res_type].append(value['name'])
+                else:
+                    perms_dict[res_type] = [value['name']]
+    for resource_type in perms_dict.keys():
+        for value in perms_dict[resource_type]:
+            permissions = entities.Permission(
+                name=value,
+                resource_type=resource_type,
+            ).search()
+        entities.Filter(permission=permissions, role=role).create()
```

Comment:
Also, how does your search work? `.search()` returns a list of results, so to take exact entity you need to use it like `.search()[0]` as otherwise it will appear in extra list (or you may even receive multiple entries with similar names if any)